pkg/generator/type_oneof.go

```diff
@@ -167,6 +187,35 @@ func (o *OneOfType) emitJSONMarshalFunc() generator.Statement {
 	)
 }
 
+func (o *OneOfType) emitToStringFunc(ctx *GeneratorContext) generator.Statement {
+	stmts := make([]generator.Statement, 0)
+	for i, alt := range o.AlternativeTypes {
+		returnStatement := generator.NewReturnStatement(`"null"`)
+
+		ts, ok := alt.(TypeWithStringConversion)
+		if ok {
+			res := fmt.Sprintf("*a.%s", ts.EmitToString(o.alternativeName(i), ctx))
+			returnStatement = generator.NewReturnStatement(res)
+		}
```

Comment:
`ts` and `ok` can be scoped to the condition:

```suggestion
		if ts, ok := alt.(TypeWithStringConversion); ok {
			res := fmt.Sprintf("*a.%s", ts.EmitToString(o.alternativeName(i), ctx))
			returnStatement = generator.NewReturnStatement(res)
		}
```